scio-core/src/main/scala/com/spotify/scio/io/FileStorage.scala

```diff
@@ -70,8 +71,11 @@ private[scio] trait FileStorage {
       .map(_.iterator().asScala).reduce(_ ++ _)
   }
 
-  def textFile: Iterator[String] =
-    IOUtils.lineIterator(getDirectoryInputStream(path), Charsets.UTF_8).asScala
+  def textFile: Iterator[String] = {
+    val in = new BufferedInputStream(getDirectoryInputStream(path))
```

Comment:
Actually this might not work for a directory of multiple compressed files since concatenated stream of gzip/bz2 != concatenated gzip/bz2 streams. That's why I added https://github.com/spotify/scio/blob/3364b1fcc003541728c42efe827407887fe48702/scio-core/src/main/scala/com/spotify/scio/io/FileStorage.scala#L127  as part of the `TFRecord` patch.